pandas/core/frame.py

```diff
@@ -4049,33 +4049,96 @@ def _compare(a, b):
 
     def combine(self, other, func, fill_value=None, overwrite=True):
         """
-        Add two DataFrame objects and do not propagate NaN values, so if for a
-        (column, time) one frame is missing a value, it will default to the
-        other frame's value (which might be NaN as well)
+        Perform series-wise combine with `other` DataFrame using given `func`.
+
+        Combines `self` DataFrame with `other` DataFrame using `func`
+        to merge columns. The row and column indexes of the resulting
+        DataFrame will be the union of the two. If `fill_value` is
```

Comment:
you are describing the parameters here (and they are described below). I think your first 2 sentences are pretty good.